The "VerifyClosure" task failed unexpectedly

It logged a "cycle" between S.Composition and itself, then threw as below. I don't know how I got into this state but it looks like it is going to happen if something is a dependency of itself
```c#
                if (depAssm == root)
                {
                    // we found a cycle for the specified root
                    var cycle = depStack.Reverse().ToArray();
                    Log.LogMessage($"Cycle detected for {depAssm.Path} : {PrintCycle(cycle)}.");

                    var suspectAssembly = cycle.First(a => a != root); // <<<<<
```
```
C:\git\corefx\pkg\frameworkPackage.targets(124,5): error MSB4018: The "VerifyClosure" task failed unexpectedly.\r [C:\g
it\corefx\pkg\Microsoft.Private.CoreFx.NETCoreApp\Microsoft.Private.CoreFx.NETCoreApp.pkgproj]
C:\git\corefx\pkg\frameworkPackage.targets(124,5): error MSB4018: System.InvalidOperationException: Sequence contains n
o matching element\r [C:\git\corefx\pkg\Microsoft.Private.CoreFx.NETCoreApp\Microsoft.Private.CoreFx.NETCoreApp.pkgproj
]
C:\git\corefx\pkg\frameworkPackage.targets(124,5): error MSB4018:    at System.Linq.Enumerable.First[TSource](IEnumerab
le`1 source, Func`2 predicate)\r [C:\git\corefx\pkg\Microsoft.Private.CoreFx.NETCoreApp\Microsoft.Private.CoreFx.NETCor
eApp.pkgproj]
C:\git\corefx\pkg\frameworkPackage.targets(124,5): error MSB4018:    at Microsoft.DotNet.Build.Tasks.Packaging.VerifyCl
osure.DumpCycles(Stack`1 depStack, AssemblyInfo root, Dictionary`2 suspectCycles)\r [C:\git\corefx\pkg\Microsoft.Privat
e.CoreFx.NETCoreApp\Microsoft.Private.CoreFx.NETCoreApp.pkgproj]
C:\git\corefx\pkg\frameworkPackage.targets(124,5): error MSB4018:    at Microsoft.DotNet.Build.Tasks.Packaging.VerifyCl
osure.DumpCycles(AssemblyInfo assembly)\r [C:\git\corefx\pkg\Microsoft.Private.CoreFx.NETCoreApp\Microsoft.Private.Core
Fx.NETCoreApp.pkgproj]
C:\git\corefx\pkg\frameworkPackage.targets(124,5): error MSB4018:    at Microsoft.DotNet.Build.Tasks.Packaging.VerifyCl
osure.Execute()\r [C:\git\corefx\pkg\Microsoft.Private.CoreFx.NETCoreApp\Microsoft.Private.CoreFx.NETCoreApp.pkgproj]
C:\git\corefx\pkg\frameworkPackage.targets(124,5): error MSB4018:    at Microsoft.Build.BackEnd.TaskExecutionHost.Micro
soft.Build.BackEnd.ITaskExecutionHost.Execute()\r [C:\git\corefx\pkg\Microsoft.Private.CoreFx.NETCoreApp\Microsoft.Priv
ate.CoreFx.NETCoreApp.pkgproj]
C:\git\corefx\pkg\frameworkPackage.targets(124,5): error MSB4018:    at Microsoft.Build.BackEnd.TaskBuilder.<ExecuteIns
tantiatedTask>d__26.MoveNext() [C:\git\corefx\pkg\Microsoft.Private.CoreFx.NETCoreApp\Microsoft.Private.CoreFx.NETCoreA
pp.pkgproj]
```

@ericstj

--- a/src/Microsoft.DotNet.Build.Tasks.Packaging/src/VerifyClosure.cs
+++ b/src/Microsoft.DotNet.Build.Tasks.Packaging/src/VerifyClosure.cs
@@ -302,7 +302,7 @@ namespace Microsoft.DotNet.Build.Tasks.Packaging
                     var cycle = depStack.Reverse().ToArray();
                     Log.LogMessage($"Cycle detected for {depAssm.Path} : {PrintCycle(cycle)}.");
 
-                    var suspectAssembly = cycle.First(a => a != root);
+                    var suspectAssembly = cycle.FirstOrDefault(a => a != root) ?? root;
 
                     AssemblyInfo[] existingCycle;
                     if (!suspectCycles.TryGetValue(suspectAssembly, out existingCycle) ||

